public/app/core/components/Login/LoginCtrl.tsx

```diff
@@ -0,0 +1,133 @@
+import React from 'react';
+import appEvents from 'app/core/app_events';
+import config from 'app/core/config';
+
+import { updateLocation } from 'app/core/actions';
+import { connect } from 'react-redux';
+import { StoreState } from 'app/types';
+import { PureComponent } from 'react';
+import { getBackendSrv } from '@grafana/runtime';
+import { hot } from 'react-hot-loader';
+
+let backendSrv;
+
+export interface FormModel {
+  user: string;
+  password: string;
+  email: string;
+}
+interface Props {
+  routeParams?: any;
+  updateLocation?: typeof updateLocation;
+  children: (obj: {
+    loggingIn: boolean;
+    changePassword: boolean;
+    toGrafana: Function;
+    signUp: Function;
+    login: (data: FormModel, valid: boolean) => void;
+  }) => JSX.Element;
+}
+
+interface State {
+  loggingIn: boolean;
+  changePassword: boolean;
```

Comment:
try to naming booleans like "isChangingPassword" , I know some if the config booleans do not follow this but as general rule this should be followed. changePassword is especially bad as it's named like an action.